fili-core/src/main/java/com/yahoo/bard/webservice/data/filterbuilders/DruidBoundFilterBuilder.java

```diff
@@ -0,0 +1,122 @@
+ *             "upper": null ,
+ *             "upperStrict": false,
+ *             "ordering": null }
+ *         }
+ *     </code></pre>
+ * </p>
+ */
+public class DruidBoundFilterBuilder implements DruidFilterBuilder {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ConjunctionDruidFilterBuilder.class);
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance();
+
+    private static final int DEFAULT_MAX_NUM_DRUID_FILTERS = 10000;
+    private static final int MAX_NUM_DRUID_FILTERS = SYSTEM_CONFIG.getIntProperty(
+            SYSTEM_CONFIG.getPackageVariableName("max_num_druid_filters"),
+            DEFAULT_MAX_NUM_DRUID_FILTERS
+    );
+
+    @Override
+    public Filter buildFilters(final Map<Dimension, Set<ApiFilter>> filterMap) throws DimensionRowNotFoundException {
+        LOG.trace("Building Druid Bound Filters using filter map: {}", filterMap);
+
+        // Assuming there is just going to be a single dimension for BOUND filter operations.
+        Dimension dimension = filterMap.keySet().iterator().next();
+        List<Filter> filters =
+                    filterMap.values()
+                            .stream()
+                            .flatMap(Set::stream)
+                            .map(filter -> buildDruidBoundFilters(filter, dimension))
+                            .collect(Collectors.toList())
+        ;
+
+        if (filters.size() == 1) {
+            return filters.get(0);
+        }
+
+        AndFilter boundFilter = new AndFilter(filters);
+
+        if (boundFilter.getFields().size() > MAX_NUM_DRUID_FILTERS) {
+            LOG.error(TOO_MANY_DRUID_FILTERS.logFormat());
+            throw new TooManyDruidFiltersException(TOO_MANY_DRUID_FILTERS.format());
+        }
+
+        LOG.trace("Filter: {}", boundFilter);
+        return boundFilter;
+    }
+
+    /**
+     * Builds a Druid Bound Filter based on the operation type.
+     *
+     * @param filter APIFilter associated to a given dimension
+     * @param dimension Dimension for the filters
+     * @return A Druid Bound Filter based on the filter operation type
+     */
+    protected Filter buildDruidBoundFilters(ApiFilter filter, Dimension dimension) {
+        List<String> values = filter.getValuesList();
+        DefaultFilterOperation filterOperation = (DefaultFilterOperation) filter.getOperation();
+        switch (filterOperation) {
+            case lte:
+                return new BoundFilter(dimension, null, values.get(0));
+            case gte:
+                return new BoundFilter(dimension, values.get(0), null);
+            case gt:
+                return new BoundFilter(dimension, values.get(0), null).withLowerBoundStrict(true);
+            case lt:
+                return new BoundFilter(dimension, null, values.get(0)).withUpperBoundStrict(true);
+            case between:
+                return new BoundFilter(dimension, values.get(0), values.get(1)).withUpperBoundStrict(true);
```

Comment:
It is hard to read what all constructor calls above do and how they are different among each other. I would suggest adding static factory methods to BoundFilter to make it much more readable.

Reply:
Could you be more elaborate? And give a sample static factory method?

Reply:
Sure.

In the case of `lte`, if you add a static factory method like this below

```java
    /**
     * 
     * @param dimension
     * @param upper
     * @return
     */
    public static BoundFilter lessThanOrEqualBound(Dimension dimension, String upper) {
        return new BoundFilter(dimension, null, upper);
    }
```

Then you could replace line https://github.com/yahoo/fili/pull/807/files/b48fea135e0d7429b6417859e80d911e77a1ba5f#diff-bf47a08b799024e899dd3860f487eb38R108 with

``` java
return BoundFilter.lessThanOrEqualBound(dimension, values.get(0));
```

This makes it obvious for read that we are creating a upper-bounded filter and we don't need to pass that mysterious `null` argument

Reply:
I don't dislike this suggestion.
I think if we were going to do this, I'd use:
BoundFilter.buildUpper(value, boolean excludeEquals)
BoundFilter.buildLower(value, boolean excludeEquals)
Then the only chain we'd need to do was fo between and we could just use the full constructor potentially for that one (with no nulls passed)